CI: `np.matrix` deprecation warning

`np.kron` is now (appeared in the NumPy pre-released job) raising a deprecation warning because we are using a matrix in `::TestSparseFunctions::test_scale_rows_and_cols`

(`np.array(*args, **kwargs).view(np.matrix)`). This is from `from scipy.sparse._sputils import matrix`.

```
/home/runner/.local/lib/python3.10/site-packages/numpy/matrixlib/defmatrix.py:116: in __new__
[1104](https://github.com/scipy/scipy/runs/5999820014?check_suite_focus=true#step:7:1104)
    warnings.warn('the matrix subclass is not the recommended way to '
[1105](https://github.com/scipy/scipy/runs/5999820014?check_suite_focus=true#step:7:1105)
E   PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
```

See for instance: https://github.com/scipy/scipy/runs/5999820014?check_suite_focus=true

--- a/scipy/sparse/tests/test_spfuncs.py
+++ b/scipy/sparse/tests/test_spfuncs.py
@@ -9,9 +9,9 @@ from scipy.sparse._sparsetools import (csr_scale_rows, csr_scale_columns,
 
 class TestSparseFunctions:
     def test_scale_rows_and_cols(self):
-        D = array([[1,0,0,2,3],
-                   [0,4,0,5,0],
-                   [0,0,6,7,0]])
+        D = array([[1, 0, 0, 2, 3],
+                   [0, 4, 0, 5, 0],
+                   [0, 0, 6, 7, 0]])
 
         #TODO expose through function
         S = csr_matrix(D)

--- a/scipy/sparse/tests/test_spfuncs.py
+++ b/scipy/sparse/tests/test_spfuncs.py
@@ -5,20 +5,19 @@ from scipy.sparse import _spfuncs as spfuncs
 from scipy.sparse import csr_matrix, csc_matrix, bsr_matrix
 from scipy.sparse._sparsetools import (csr_scale_rows, csr_scale_columns,
                                        bsr_scale_rows, bsr_scale_columns)
-from scipy.sparse._sputils import matrix
 
 
 class TestSparseFunctions:
     def test_scale_rows_and_cols(self):
-        D = matrix([[1,0,0,2,3],
-                    [0,4,0,5,0],
-                    [0,0,6,7,0]])
+        D = array([[1,0,0,2,3],
+                   [0,4,0,5,0],
+                   [0,0,6,7,0]])
 
         #TODO expose through function
         S = csr_matrix(D)
         v = array([1,2,3])
         csr_scale_rows(3,5,S.indptr,S.indices,S.data,v)
-        assert_equal(S.toarray(), diag(v)*D)
+        assert_equal(S.toarray(), diag(v)@D)
 
         S = csr_matrix(D)
         v = array([1,2,3,4,5])

